pallets/loans/src/lib.rs

```diff
@@ -916,16 +914,16 @@ impl<T: Config> Pallet<T> {
     fn redeem_allowed(
         currency_id: &CurrencyId,
         redeemer: &T::AccountId,
-        redeem_amount: Balance,
-    ) -> DispatchResult {
+        voucher_amount: Balance,
+    ) -> Result<Balance, DispatchError> {
         let deposit = Self::account_deposits(currency_id, redeemer);
         let exchange_rate = Self::exchange_rate(currency_id);
-        let voucher_amount = Self::calc_collateral_amount(redeem_amount, exchange_rate)?;
+        let redeem_amount = Self::calc_underlying_amount(voucher_amount, exchange_rate)?;
```

Comment:
redeem_amount has no usage in this function.

Reply:
it's used in `redeem_internal`

Reply:
That should be calculated in `redeem_internal`.

Reply:
@yz89 as you can see from here: https://github.com/parallel-finance/parallel/pull/249/commits/a5f7ee0fba725cd5315ed5b248c5f9f7897641bd

if I remove the return value, there will be some unecessary calculations